### PR TITLE
Add ability to escape variable expansion in policy

### LINKED ISSUE #2177: No way to escape variables in transpiler parsing
There is no easy way to provide a raw `${var}` to a computed unit configuration. At the moment the transpiler will determine that to be a variable and try to replace it, which could be something that the unit configuration would not want.

We should support the usage of `$${var}`, where the transpiler just ignores it and removes the initial `$` resulting in the unit configuration get the constant value of `${var}`.

At the moment the workaround is to use the pipe `|` to work around this and provide a default.

`${vars|'$vars'}`

This is not a great user experience and also has an issue where `vars` must also not exist or it will do the wrong thing.

For confirmed bugs, please report:
- Version: 8.6.0
- Operating System: All

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds the ability to a variable not be expanded and replaced in inputs.

A variable that is defined as `$${...}` will be replaced with `${...}` in the policy.

`$${...}` is not allowed in any `condition` statement, because a condition statement requires that strings always be wrapped in `'...'` and so it doesn't make sense to allow it to be escaped and converted into a real string. That is already supported by just doing `'${...}'`.

## Why is it important?

Allows users to provide values that should not be replaced in the policy.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- ~~[ ] I have added an integration test or an E2E test~~(unit test coverage is really good)

## Disruptive User Impact

Adds the ability of users to escape variables.

## Related issues

- Closes #2177

### CODE REVIEW

**Comment by blakerouse** [comment]
Seems SonarQube is mad about the generated code from antlr, not from the code added.

**Comment by pierrehilbert** [comment]
I'm okay with force merging this if we are getting the approval from code reviewer, @ycombinator is probably sharing my thoughts here

File: internal/pkg/eql/visitor.go [hunk 1]
```diff
@@ -250,6 +250,13 @@ func (v *expVisitor) VisitExpNot(ctx *parser.ExpNotContext) interface{} {
 	return !val
 }
 
+func (v *expVisitor) VisitExpEVariable(ctx *parser.ExpEVariableContext) interface{} {
+	// escaped variables are not allowed in Eql conditions
+	// they should be wrapped in a constant to be allowed as a valid string in the expression
+	v.err = errors.New("escaped variable $${ is not allowed")
```
Comment (michel-laterman) [inline internal/pkg/eql/visitor.go:256]: (nitpick) Would this error message be clear in the logs that this is not allowed in a condition?

**Review by michel-laterman — APPROVED** [review]
lgtm

File: internal/pkg/eql/visitor.go (hunk 1 above)
Comment (blakerouse) [reply internal/pkg/eql/visitor.go:256]: Yes the condition checker adds that context.

**Comment by ycombinator** [comment]
Force merging as this PR contains a lot of generated code and SonarQube is not able to exclude it.

**Pull request merged**